docs: mention s3 and audit log storage in etcd backend (#28985)
gravitational/teleport: docs/pages/reference/backends.mdx

    etcd](https://github.com/etcd-io/etcd/tree/master/hack/tls-setup) if you
    don't already have a TLS setup.
- Configure all Teleport Auth servers to use etcd in the "storage" section of the config file as shown below.

> - Each Teleport Auth server will have individual audit logs within the `log` directory under the data

Reviewer: ```suggestion
- Each Teleport Auth Service instance will have individual audit logs within the `log` directory under the data
```

>   directory (by default `/var/lib/teleport/log`). Using a SIEM to consolidate audit logs is recommended.

Reviewer: If we're going to recommend using a SIEM to consolidate audit logs, we might want to link to the section of the docs re: setting up the Event Handler for different services.

>   Using a S3 compatible storage will allow for a central storage point of the sessions. Otherwise each

Reviewer: ```suggestion
  Using an S3-compatible storage solution will allow for a central platform for storing sessions. Otherwise each
```

>   Teleport Auth server will have sessions stored in the `records` directory under the same `log` directory 

Reviewer: ```suggestion
  Teleport Auth Service instance will have sessions stored in the `records` directory under the same `log` directory 
```

  (by default `/var/lib/teleport/log/records`). See the S3 section.

> - Deploy several auth servers connected to etcd backend.

Reviewer: While we're here

```suggestion
- Deploy several Auth Service instances connected to an etcd backend.
```

  in Teleport Architecture documentation.
</Admonition>

> S3 and S3 compatible buckets can only be used as storage for the recorded sessions. S3 cannot

Reviewer: ```suggestion
S3 and S3-compatible buckets can only be used as storage for the recorded sessions. S3 cannot
```

> store the audit log or the cluster state.

Reviewer 2: Note, this is technically no longer true, as we now have an S3 based audit log too.
